sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -210,12 +211,16 @@ def fetch_propagator(name, class_name, gem_suffix = name)
         nil
       end
 
-      def fetch_exporter(name, class_name)
-        Trace::Export::BatchSpanProcessor.new(Kernel.const_get(class_name).new)
+      def fetch_exporter(name, class_name, exporter_options = {})
+        Trace::Export::BatchSpanProcessor.new(Kernel.const_get(class_name).new(**exporter_options), metrics_reporter: metrics_reporter)
       rescue NameError
         OpenTelemetry.logger.warn "The #{name} exporter cannot be configured - please add opentelemetry-exporter-#{name} to your Gemfile, spans will not be exported"
         nil
       end
+
+      def metrics_reporter
+        @trace_metrics_reporter_provider.call
```

Comment:
Possible i'm missing some logic in how @trace_metrics_reporter_provider gets defined, but iiuc, if @ `trace_metrics_reporter_provider` is not set and defaults to nil, wouldn't this throw an error?
```suggestion
        @trace_metrics_reporter_provider.call unless @trace_metrics_reporter_provider.nil?
```

Reply:
It defaults to a proc that returns a `nil` metrics reporter:

https://github.com/open-telemetry/opentelemetry-ruby/pull/1110/files#diff-e6a05d2b9d8d72a4c1ab589e75a9aa34e484e073af6443d428929c0c554f93eaR44

and in turn BSP and OTLP exporter default to a no-op implementation:

https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb#L66

https://github.com/open-telemetry/opentelemetry-ruby/blob/main/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L73